overlord/export_test.go

```diff
@@ -0,0 +1,31 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package overlord
+
+import (
+	"time"
+)
+
+// SetEnsureIntervalForTest let's change overlord ensure interval for tests.
+func SetEnsureIntervalForTest(d time.Duration) time.Duration {
```

Comment:
That's what I used to do in my test hacks as well, but I ended up finding a slightly different version much nicer to use:

`func SetEnsureIntervalForTest(d time.Duration) (restore func()) {`

That is, instead of returning the old, return a function that puts the original old back in place. So the call sites become
```
restore := HackFooBar()
defer restore()
```